modules/carto/src/sources/index.ts

```diff
@@ -0,0 +1,33 @@
+export {SOURCE_DEFAULTS} from './common';
+export type {
+  CartoSourceOptionalOptions,
```

Comment:
CartoSourceOptionalParams / CartoSourceRequiredParams ? to avoid the cacophony?

Reply:
Well they are options rather than parameters if you go by the `fn(param1, param2)` vs `fn({option1, option2})` convention. I'm actually not going to export these types in the end, they are mainly there internally to help distinguish between options for which the client has defaults and those that mush be provided by the user